cobalt/updater/configurator.cc

```diff
@@ -0,0 +1,422 @@
+const char kUpdaterJSONDefaultUrlQA[] =
+    "https://omaha-qa.sandbox.google.com/service/update2/json";
+const char kUpdaterJSONDefaultUrl[] =
+    "https://tools.google.com/service/update2/json";
+
+std::string GetDeviceProperty(SbSystemPropertyId id) {
+  char value[kSystemPropertyMaxLength];
+  bool result;
+  result = SbSystemGetProperty(id, value, kSystemPropertyMaxLength);
+  std::string prop;
+  if (result) {
+    prop = std::string(value);
+  }
+  return prop;
+}
+
+}  // namespace
+
+namespace cobalt {
+namespace updater {
+
+Configurator::Configurator(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : pref_service_(CreatePrefService()),
+      persisted_data_(
+          std::make_unique<update_client::PersistedData>(pref_service_.get(),
+                                                         nullptr)),
+      network_fetcher_factory_(
+          base::MakeRefCounted<NetworkFetcherFactoryCobalt>(
+              url_loader_factory)),
+      crx_downloader_factory_(
+          update_client::MakeCrxDownloaderFactory(network_fetcher_factory_)),
+      unzip_factory_(base::MakeRefCounted<UnzipperFactory>()),
+      patch_factory_(base::MakeRefCounted<PatcherFactory>()),
+      is_forced_update_(0),
+      allow_self_signed_packages_(false),
+      require_network_encryption_(true) {
+  LOG(INFO) << "Configurator::Configurator";
+
+  const std::string persisted_channel = persisted_data_->GetLatestChannel();
+  if (persisted_channel.empty()) {
+    SetChannel(kDefaultUpdaterChannel);
+  } else {
+    SetChannel(persisted_channel);
+  }
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+          switches::kUseUncompressedUpdates)) {
+    use_compressed_updates_.store(false);
+  } else {
+    use_compressed_updates_.store(true);
+  }
+}
+Configurator::~Configurator() {
+  LOG(INFO) << "Configurator::~Configurator";
+}
+
+base::TimeDelta Configurator::InitialDelay() const {
+  return base::Seconds(0);
+}
+
+base::TimeDelta Configurator::NextCheckDelay() const {
+  return base::Hours(5);
+}
+
+base::TimeDelta Configurator::OnDemandDelay() const {
+  return base::Seconds(0);
+}
+
+base::TimeDelta Configurator::UpdateDelay() const {
+  return base::Seconds(0);
+}
+
+std::vector<GURL> Configurator::UpdateUrl() const {
+#if !defined(COBALT_BUILD_TYPE_GOLD)
+  if (allow_self_signed_packages_ && !update_server_url_.empty()) {
+    return std::vector<GURL>{GURL(update_server_url_)};
+  }
+#endif  // !defined(COBALT_BUILD_TYPE_GOLD)
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+          switches::kUseQAUpdateServer)) {
+    return std::vector<GURL>{GURL(kUpdaterJSONDefaultUrlQA)};
+  } else {
+    return std::vector<GURL>{GURL(kUpdaterJSONDefaultUrl)};
+  }
+}
+
+std::vector<GURL> Configurator::PingUrl() const {
+  return UpdateUrl();
+}
+
+std::string Configurator::GetProdId() const {
+  return "cobalt";
+}
+
+base::Version Configurator::GetBrowserVersion() const {
```

Comment:
nit: Can we just return `COBALT_MAJOR_VERSION`?

https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/main:cobalt/version.h;l=39;drc=1a248c495379f69283e4f19a23a2ccba141d90c3